app/src/components/generic/backButton/BackButton.test.tsx

```diff
@@ -23,4 +26,64 @@ describe('BackButton', () => {
             expect(history.location.pathname).toBe('/');
         });
     });
+
+    it('calls the login handler when clicking the back button on the upload search page', async () => {
+        const test_location_prefix = useBaseAPIUrl();
```

Comment:
```
import { useBaseAPIUrl } from '../../../providers/configProvider/ConfigProvider';
jest.mock('../../../providers/configProvider/ConfigProvider');
const mockUseBaseAPIUrl = useBaseAPIUrl as jest.Mock;

const testUrl = '/test'
...

                mockUseBaseAPIUrl.mockReturnValue(testUrl);
                expect(window.location.replace).toBeCalledWith(testUrl + endpoints.LOGIN);
```

We should mock dependencies and replace  the actual implementation with a test implementation.This allows us to test the links between code, the input and the output expectations. Rather than use the code itself to reach an outcome.